src/stream/src/executor/backfill/snapshot_backfill/executor.rs

```diff
@@ -169,14 +169,22 @@ impl<S: StateStore> SnapshotBackfillExecutor<S> {
                     .snapshot_backfill_consume_row_count
                     .with_guarded_label_values(&[&table_id_str, &actor_id_str, "consume_upstream"]);
 
-                let mut upstream_buffer =
-                    UpstreamBuffer::new(&mut self.upstream, consume_upstream_row_count);
-
-                let first_barrier_epoch = first_upstream_barrier.epoch;
-                let snapshot_epoch = first_barrier_epoch.prev;
+                let mut upstream_buffer = UpstreamBuffer::new(
+                    &mut self.upstream,
+                    first_upstream_barrier,
+                    consume_upstream_row_count,
+                );
 
                 // Phase 1: consume upstream snapshot
+                let (mut barrier_epoch, upstream_buffer) = if first_recv_barrier_epoch.prev
+                    < snapshot_epoch
```

Comment:
`first_recv_barrier_epoch` should be the latest epoch of upstream right? Shouldn't it be:

`first_recv_barrier_epoch > snapshot_epoch`

Reply:
Oh nevermind, it's actually the _second_ barrier received from upstream. I suppose it's the state table epoch.